src/OSPSuite.Core/Domain/Services/SimModelBatch.cs

```diff
@@ -20,17 +20,59 @@ public class SimModelBatch : SimModelManagerBase, ISimModelBatch
       private IReadOnlyList<SpeciesProperties> _allVariableMolecules;
       private string _simulationResultsName;
       private bool _calculateSensitivities;
+      private ICloneManagerForModel _cloneManagerForModel;
+      private ISimModelSimulationFactory _simModelSimulationFactory;
 
-      public SimModelBatch(ISimModelExporter simModelExporter, ISimModelSimulationFactory simModelSimulationFactory, IDataFactory dataFactory) : base(
+      public SimModelBatch(ISimModelExporter simModelExporter, ISimModelSimulationFactory simModelSimulationFactory, IDataFactory dataFactory, ICloneManagerForModel cloneManagerForModel) : base(
          simModelExporter, simModelSimulationFactory)
       {
          _dataFactory = dataFactory;
+         _simModelExporter = simModelExporter;
+         _simModelSimulationFactory = simModelSimulationFactory;
+         _cloneManagerForModel = cloneManagerForModel;
       }
 
       public IReadOnlyList<string> VariableParameterPaths { get; private set; }
 
       public IReadOnlyList<string> VariableMoleculePaths { get; private set; }
 
+      public object Clone()
+      {
+         var xml = _simModelExporter.ExportSimModelXml(_modelCoreSimulation, new SimulationRunOptions().SimModelExportMode);
```

Comment:
wow.. what's going on here?😳
What are you trying to achieve? If feels like you are  not working at the right abstraction level here